state/remoteapplication.go

```diff
@@ -1024,19 +1044,53 @@ func (st *State) AddRemoteApplication(args AddRemoteApplicationParams) (_ *Remot
 				return nil, errors.AlreadyExistsf("saas application")
 			}
 		}
-		ops := []txn.Op{
-			model.assertActiveOp(),
-			{
+
+		var ops []txn.Op
+
+		// If there's an older existing consumer app proxy for the same consuming app,
+		// it needs to be removed.
+		if destroyStaleOp, err = st.destroyExistingRemoteApplicationOp(args); err != nil {
+			return nil, errors.Trace(err)
+		}
+		replaceExisting := destroyStaleOp != nil
+		if replaceExisting {
+			destroyOps, err := destroyStaleOp.Build(attempt)
+			if err != nil {
+				return nil, errors.Trace(err)
+			}
```

Comment:
As discussed, we need an assertion here to confirm that the existing remote app exists with the version we read outside the txn.